src/Microsoft.ML.PCA/PcaTrainer.cs

```diff
@@ -62,11 +62,14 @@ public class Arguments : LearnerInputBaseWithWeight
             public int Oversampling = 20;
 
             [Argument(ArgumentType.AtMostOnce, HelpText = "If enabled, data is centered to be zero mean", ShortName = "center")]
-            [TlcModule.SweepableDiscreteParam("Center", null, isBool:true)]
+            [TlcModule.SweepableDiscreteParam("Center", null, isBool: true)]
             public bool Center = true;
 
             [Argument(ArgumentType.AtMostOnce, HelpText = "The seed for random number generation", ShortName = "seed")]
             public int? Seed;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Column to use for example weight", ShortName = "weight", SortOrder = 4, Visibility = ArgumentAttribute.VisibilityType.EntryPointsOnly)]
+            public Optional<string> WeightColumn = Optional<string>.Implicit(DefaultColumnNames.Weight);
```

Comment:
Hmmm. Actually this is interesting. I see you're adding this here, but K-means definitely supports weighted training, descends from this same abstract class, `LearnerInputBase`, and did not specify a weight column input. :( I'll file a separate issue for that.